aws-api-appsync/src/main/java/com/amplifyframework/datastore/appsync/ModelConverter.java

```diff
@@ -45,41 +45,33 @@ private ModelConverter() {}
-                            && !((SerializedModel) instance).getSerializedData().containsKey(modelField.getName())) {
-                        // Skip fields that are not set, so that they are not set to null in the request.
-                        continue;
-                    }
-                    result.put(fieldName, extractFieldValue(modelField, instance));
-                } else if (association.isOwner()) {
-                    Object associateId = extractAssociateId(modelField, instance);
-                    if (associateId == null) {
-                        // Skip fields that are not set, so that they are not set to null in the request.
-                        continue;
-                    }
-                    result.put(fieldName, SerializedModel.builder()
-                        .serializedData(Collections.singletonMap("id", associateId))
-                        .modelSchema(null)
-                        .build());
+            final ModelAssociation association = schema.getAssociations().get(fieldName);
+            if (association == null) {
+                if (instance instanceof SerializedModel
+                        && !((SerializedModel) instance).getSerializedData().containsKey(modelField.getName())) {
+                    // Skip fields that are not set, so that they are not set to null in the request.
+                    continue;
+                }
+                result.put(fieldName, extractFieldValue(modelField.getName(), instance, schema));
+            } else if (association.isOwner()) {
+                Object associateId = extractAssociateId(modelField, instance, schema);
+                if (associateId == null) {
+                    // Skip fields that are not set, so that they are not set to null in the request.
+                    continue;
                 }
-                // Ignore if field is associated, but is not a "belongsTo" relationship
-            } catch (Exception exception) {
-                throw new AmplifyException(
-                        "An invalid field was provided. " + fieldName + " is not present in " + schema.getName(),
-                        exception,
-                        "Check if this model schema is a correct representation of the fields in the provided Object");
+                result.put(fieldName, SerializedModel.builder()
+                    .serializedData(Collections.singletonMap("id", associateId))
+                    .modelSchema(null)
+                    .build());
             }
+            // Ignore if field is associated, but is not a "belongsTo" relationship
         }
         return result;
     }
 
-    private static Object extractAssociateId(ModelField modelField, Model instance)
-            throws NoSuchFieldException, IllegalAccessException {
-        final Object fieldValue = extractFieldValue(modelField, instance);
+    private static Object extractAssociateId(ModelField modelField, Model instance, ModelSchema schema)
```

Comment:
No functional changes were made in this file.  I just did a little refactoring to make `extractAssociateId` and `extractFieldValue` throw an `AmplifyException` instead of `NoSuchFieldException, and `IllegalAccessException`.

These changes were made in preparation for a larger simplification that I plan to do in a separate PR, to DRY up some of this logic which is duplicated in `ModelConverter`, `AppSyncRequestFactory`, and `AppSyncGraphQLRequestFactory`